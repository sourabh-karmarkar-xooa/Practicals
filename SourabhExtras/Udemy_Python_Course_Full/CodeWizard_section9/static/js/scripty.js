$(document).ready(function(){
    console.log("loaded")
    $.material.init();
    $(document).on("submit","#register-form", function(e){
        e.preventDefault();
        console.log("form submit")
    })
})

