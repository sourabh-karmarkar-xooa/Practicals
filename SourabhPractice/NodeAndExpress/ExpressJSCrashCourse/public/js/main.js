$(document).ready(function(){
    $('.deleteUser').on('click',deleteUser);
});

function deleteUser(){
     var confirmation=confirm('Are You Sure?');
     if(confirmation)
     {
         $.ajax({
             type:'DELETE',
             url:'/users/delete/'+$(this).data('id')
         }).done(function(){
            window.location.replace('/');
         });
         window.location.replace('/');
     }
     else
     {
         return false;
     }
}