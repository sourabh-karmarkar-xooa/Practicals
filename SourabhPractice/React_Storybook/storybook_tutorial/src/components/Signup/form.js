import React from "react";
import PropTypes from "prop-types";
import {
  Wrapper,
  Form,
  HeaderRow,
  ActionRow,
  Button,
  SubmitButton
} from "../../styles";
import { HorizontalInputField } from "../HorizontalInputField";

/**
 * SignupForm - presents and manages a signup form for a web site
 *
 * @class
 * @param {object} props Properties passed to the instance of the class
 * @property {string} [title] Title to use for form
 * @property {object} [account] Prior account data
 * @property {function} onSubmit Callback to recieve the form data object when the submit button is clicked
 * @property {function} onCancel Callback invoked when the Cancel button is clicked
 */
export class SignupForm extends React.Component {
  static propTypes = {
    title: PropTypes.string,
    account: PropTypes.object,
    onSubmit: PropTypes.func.isRequired,
    onCancel: PropTypes.func.isRequired
  };

  static defaultProps = {
    title: "Sign up",
    account: {}
  };

  state = {
    email: "",
    password: "",
    passwordConfirmation: "",
    phone: ""
  };

  handleChange = e => {
    const target = e.target;
    const name = target.name;
    const value = target.value;
    this.setState({
      [name]: value
    });
  };

  handleCancel = () => {
    this.props.onCancel();
  };

  handleSubmit = () => {
    this.props.onSubmit(this.state);
  };

  componentDidMount() {
    if (this.props.account) {
      const email = this.props.account.email || "";
      const phone = this.props.account.phone || "";
      this.setState({ email, phone });
    }
  }

  render() {
    return (
      <Wrapper>
        <Form>
          <HeaderRow>
            <h3>{this.props.title}</h3>
          </HeaderRow>
          <HorizontalInputField
            id="email-field"
            name="email"
            label="Email:"
            value={this.state.email}
            onChange={this.handleChange}
          />
          <HorizontalInputField
            id="password-field"
            name="password"
            label="Password:"
            value={this.state.password}
            onChange={this.handleChange}
          />
          <HorizontalInputField
            id="password-confirmation-field"
            name="passwordConfirmation"
            label="Confirm Password:"
            value={this.state.passwordConfirmation}
            onChange={this.handleChange}
          />
          <HorizontalInputField
            id="phone-field"
            name="phone"
            label="Phone:"
            value={this.state.phone}
            onChange={this.handleChange}
          />
          <ActionRow>
            <Button type="button" onClick={this.handleCancel}>
              Cancel
            </Button>
            <SubmitButton type="button" onClick={this.handleSubmit}>
              Save
            </SubmitButton>
          </ActionRow>
        </Form>
      </Wrapper>
    );
  }
}
export default SignupForm;
