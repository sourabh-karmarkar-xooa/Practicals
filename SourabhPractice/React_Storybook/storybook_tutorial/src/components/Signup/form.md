<a name="SignupForm"></a>

## SignupForm
SignupForm - presents and manages a signup form for a web site

**Kind**: global class  
**Properties**

| Name | Type | Description |
| --- | --- | --- |
| [title] | <code>string</code> | Title to use for form |
| [account] | <code>object</code> | Prior account data |
| onSubmit | <code>function</code> | Callback to recieve the form data object when the submit button is clicked |
| onCancel | <code>function</code> | Callback invoked when the Cancel button is clicked |

<a name="new_SignupForm_new"></a>

### new exports.SignupForm(props)

| Param | Type | Description |
| --- | --- | --- |
| props | <code>object</code> | Properties passed to the instance of the class |

