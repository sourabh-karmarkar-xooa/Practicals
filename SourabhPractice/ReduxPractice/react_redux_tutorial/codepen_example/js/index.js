const {createStore} = Redux;

const initstate={
    todos:[],
    posts:[]
}

function myreducer(state=initstate,action){
    if(action.type=='ADD_TODO')
    {
        return {
            ...state,
            todos:[...state.todos,action.todo]
        }
    }
    if(action.type=='ADD_POST')
    {
        return {
            ...state,
            posts:[...state.posts,action.post]
        }
    }
    console.log(action);
    console.log(state);
}

const store=createStore(myreducer);
store.subscribe(()=>{
    console.log('state updated');
    console.log(store.getState());
});

store.dispatch({type:'ADD_TODO',todo:'Buy milk'});
store.dispatch({type:'ADD_TODO',todo:'Sleep Some More'});
store.dispatch({type:'ADD_POST',post:'Egg Hunt'});

