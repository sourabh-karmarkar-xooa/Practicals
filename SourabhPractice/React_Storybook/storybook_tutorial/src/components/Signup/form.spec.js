import React from "react";
import renderer from "react-test-renderer";
import { SignupForm } from "./form.js";

describe("<SignupForm /> test", () => {
  it("renders the component", () => {
    const actual = renderer.create(
      <SignupForm onSubmit={jest.fn()} onCancel={jest.fn()} />
    );
    expect(actual).toMatchSnapshot();
  });
  it("renders with existing account", () => {
    const account = {
      email: "test@email.com",
      phone: "555-1212"
    };
    const actual = renderer.create(
      <SignupForm
        title="Edit account"
        account={account}
        onSubmit={jest.fn()}
        onCancel={jest.fn()}
      />
    );
    expect(actual).toMatchSnapshot();
    const inputs = actual.root.findAllByType("input");
    expect(Array.isArray(inputs)).toBeTruthy();
    const values = inputs.reduce((valuesHash, input) => {
      if (input.props.value) {
        valuesHash[input.props.name] = input.props.value;
      }
      return valuesHash;
    }, {});
    expect(values).toEqual(account);
  });
});
