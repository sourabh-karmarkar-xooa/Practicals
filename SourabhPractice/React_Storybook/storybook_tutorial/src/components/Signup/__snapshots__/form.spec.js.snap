// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupForm /> test renders the component 1`] = `
<div
  className="sc-bdVaJa foQXQy"
>
  <div
    className="sc-bwzfXH lopfab"
  >
    <header
      className="sc-bxivhb dWQBoy"
    >
      <h3>
        Sign up
      </h3>
    </header>
    <div
      className="sc-htpNat iAbEgG"
    >
      <label
        htmlFor="email"
      >
        Email:
      </label>
      <input
        className="form-control"
        id="email-field"
        name="email"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="sc-htpNat iAbEgG"
    >
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="password-field"
        name="password"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="sc-htpNat iAbEgG"
    >
      <label
        htmlFor="passwordConfirmation"
      >
        Confirm Password:
      </label>
      <input
        className="form-control"
        id="password-confirmation-field"
        name="passwordConfirmation"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="sc-htpNat iAbEgG"
    >
      <label
        htmlFor="phone"
      >
        Phone:
      </label>
      <input
        className="form-control"
        id="phone-field"
        name="phone"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="sc-ifAKCX gwtZdg"
    >
      <button
        className="sc-EHOje jJiyEz"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="sc-bZQynM jeGcWw"
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;

exports[`<SignupForm /> test renders with existing account 1`] = `
<div
  className="sc-bdVaJa foQXQy"
>
  <div
    className="sc-bwzfXH lopfab"
  >
    <header
      className="sc-bxivhb dWQBoy"
    >
      <h3>
        Edit account
      </h3>
    </header>
    <div
      className="sc-htpNat iAbEgG"
    >
      <label
        htmlFor="email"
      >
        Email:
      </label>
      <input
        className="form-control"
        id="email-field"
        name="email"
        onChange={[Function]}
        value="test@email.com"
      />
    </div>
    <div
      className="sc-htpNat iAbEgG"
    >
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        className="form-control"
        id="password-field"
        name="password"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="sc-htpNat iAbEgG"
    >
      <label
        htmlFor="passwordConfirmation"
      >
        Confirm Password:
      </label>
      <input
        className="form-control"
        id="password-confirmation-field"
        name="passwordConfirmation"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="sc-htpNat iAbEgG"
    >
      <label
        htmlFor="phone"
      >
        Phone:
      </label>
      <input
        className="form-control"
        id="phone-field"
        name="phone"
        onChange={[Function]}
        value="555-1212"
      />
    </div>
    <div
      className="sc-ifAKCX gwtZdg"
    >
      <button
        className="sc-EHOje jJiyEz"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="sc-bZQynM jeGcWw"
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
